x-pack/libbeat/common/aws/credentials.go

```diff
@@ -45,6 +46,16 @@ func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
 	// Please see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
 	// with more details.
 	if config.ProfileName != "" {
+		if config.SharedCredentialFile != "" {
+			// If shared_credential_file is empty, then external.LoadDefaultAWSConfig
+			// function will load AWS config from current user's home directory.
+			// Linux/OSX: "$HOME/.aws/credentials"
+			// Windows:   "%USERPROFILE%\.aws\credentials"
+			return external.LoadDefaultAWSConfig(
+				external.WithSharedConfigProfile(config.ProfileName),
+				external.WithSharedConfigFiles([]string{config.SharedCredentialFile}),
+			)
+		}
```

Comment:
To avoid nesting ifs for these options you can use an array and have an only final call to `LoadDefaultAWSConfig`
```
options := []external.Config

if config.ProfileName != "" {
  options = append(options, external.WithSharedConfigProfile(config.ProfileName)
}
if #config.SharedCredentialFile != "" {
  options = append(options, external.WithSharedConfigFiles([]string{config.SharedCredentialFile})
}

return external.LoadDefaultAWSConfig(options...)
```